tests/integration/test_notes.py

```diff
@@ -28,3 +29,19 @@ def test_it_returns_404_with_invalid_ticket(self, valid_header):
             headers=valid_header,
         )
         assert is_valid(response, 404)  # Bad Request- 'Invalid Ticket'
+
+    def test_it_deletes_one_note(
+        self, valid_header, create_note, create_admin_user, create_ticket
+    ):
+        note = create_note(create_admin_user(), create_ticket())
+
+        response = self.client.delete(
+            f"{self.endpoint}/{note.ticketid}/notes/{note.id}", headers=valid_header
+        )
+
+        assert response.json == {"message": "Note deleted"}
```

Comment:
FYI: If I were to remove the following line from the Note endpoint under test.
`NotesModel.delete(note_id)`
Then this test would still pass. We want to test behavior here. Behavior could be stated as follows:
> The delete endpoint removes a Note resource, identified by the provided id, from the database.

The preferred way to test this is to test that the `NotesModel.delete` method is being called with the correct arguments. 

Another way to test this is to check if the created Note no longer exists in the database, but that's technically retesting behavior that is already well tested.

Reply:
I understand what you are saying, and it makes sense.  The way it is currently setup, this essentially tests that no errors are thrown, rather than NotesModel.delete is called successfully.  That being said, I'm not sure how to test NotesModel.delete when it doesn't return anything, and without testing that the note can no longer be found.

We could change the BaseModel.delete to return a success message, though I'm not sure what the implications of that would be for the rest of the models.  

Is there better way to test this?

Reply:
Here is a code example of how to test a method gets called correctly.
https://github.com/codeforpdx/dwellinglybackend/blob/273521e410b0a3c15359b8a9a2e0623d84df712d/tests/integration/test_leases.py#L52-L58

Note: You can literally pass any value as the argument, but I'd probably prefer an actual note id created through a factory is used. Which this example doesn't do.

We're basically stubbing out the model and database here. We know the delete method works, and we know the database works. We just want to assert the endpoint calls the delete method with the correct argument. And I believe we do have a true system test somewhere just in case anything does change with the model/database in the future, so I'm pretty comfortable with this level of testing and nothing more is needed here.

Reply:
I reworked the test based on test_leases.   Does this look better?

 I used the create_note factory.  For that reason we will want to remove the arguments being passed when #283 is merged in.

Reply:
yep, Looks great. I like this because now we'll know if anything changes in the endpoint that causes the delete method to not get called.